src/datasets/io/json.py

```diff
@@ -81,9 +82,13 @@ def write(self) -> int:
         _ = self.to_json_kwargs.pop("path_or_buf", None)
         orient = self.to_json_kwargs.pop("orient", "records")
         lines = self.to_json_kwargs.pop("lines", True)
+        compression = self.to_json_kwargs.pop("compression", None)
+
+        if compression not in [None, "gzip"]:
+            raise NotImplementedError(f"Datasets currently does not support {compression} compression")
```

Comment:
we use backticks to refer to the library
```suggestion
            raise NotImplementedError(f"`datasets` currently does not support {compression} compression")
```